drivers/led/ht16k33.c

```diff
@@ -135,11 +135,6 @@ static int ht16k33_led_set_brightness(const struct device *dev, uint32_t led,
 	uint8_t dim;
 	uint8_t cmd;
 
-	if (value < dev_data->min_brightness ||
-	    value > dev_data->max_brightness) {
```

Comment:
Maybe we could push it one bridge further and kill the the `{min,max}_brightness` fields of `struct led_data` structure ? I don't think there are any users left... I think (not sure) they were there for LED drivers to store their hardware limits, for internal use. But for some reason all LED drivers used these fields to store the API brightness range.

And maybe, this would be the opportunity to completely kill the whole `struct led_data` structure. The `{min,max}_period` fields could be moved o the driver's private data structure.

Reply:
yeah had a half thought about it but I haven't looked in the details, can we do it in a followup?

Reply:
Of course we can ! It is already in a much better state thanks to your efforts.